From 28ef23a7a889df84da6df481249d39bb4dddb4f9 Mon Sep 17 00:00:00 2001
From: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
Date: Mon, 23 Aug 2010 15:23:43 -0700
Subject: [PATCH 01/26] xen/apic: use handle_edge_irq for pirq events

Like other events, pirq events are edge-triggered.

Signed-off-by: Jeremy Fitzhardinge <jeremy.fitzhardinge@citrix.com>
(cherry picked from commit 29a2e2a7bd19233c62461b104c69233f15ce99ec)

Signed-off-by: Ian Campbell <ijc@hellion.org.uk>
---
 drivers/xen/events.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/xen/events.c b/drivers/xen/events.c
index 904b402..ae82e08 100644
--- a/drivers/xen/events.c
+++ b/drivers/xen/events.c
@@ -598,7 +598,7 @@ int xen_allocate_pirq(unsigned gsi, int shareable, char *name)
 		irq = find_unbound_irq();
 
 	set_irq_chip_and_handler_name(irq, &xen_pirq_chip,
-				      handle_level_irq, name);
+				      handle_edge_irq, name);
 
 	irq_op.irq = gsi;
 	irq_op.vector = 0;
@@ -707,7 +707,7 @@ int xen_create_msi_irq(struct pci_dev *dev, struct msi_desc *msidesc, int type)
 		irq_info[irq].u.pirq.domid = domid;
 
 	set_irq_chip_and_handler_name(irq, &xen_pirq_chip,
-			handle_level_irq,
+			handle_edge_irq,
 			(type == PCI_CAP_ID_MSIX) ? "msi-x":"msi");
 
 out:
-- 
1.7.5.4

